app/services/filter_tagged_content.rb

```diff
@@ -0,0 +1,62 @@
+class FilterTaggedContent
+  def valid?(document)
+    return valid_document_collection?(document) if document.document_collection?
+
+    valid_content_item?(document)
+  end
+
+private
+
+  def valid_document_collection?(document)
```

Comment:
That's a nice way of writing it (I've done it quite a bit in the past), but perhaps a bit less explicit, which might lead to confusion.